torch_geometric/nn/dense/linear.py

```diff
@@ -208,10 +214,21 @@ def forward(self, x: Tensor, type_vec: Tensor) -> Tensor:
             x (Tensor): The input features.
             type_vec (LongTensor): A vector that maps each entry to a type.
         """
-        out = x.new_empty(x.size(0), self.out_channels)
-        for i, lin in enumerate(self.lins):
-            mask = type_vec == i
-            out[mask] = lin(x[mask])
+        if self._WITH_PYG_LIB:
+            if not self.is_sorted:
+                if (type_vec[1:] < type_vec[:-1]).any():
+                    type_vec, perm = type_vec.sort()
+                    x = x[:, perm]
+            edge_type_ptr = torch.ops.torch_sparse.ind2ptr(
+                type_vec, self.num_types)
+            out = segment_matmul(x, edge_type_ptr, self.weights)
+            for i in range(self.num_types):
```

Comment:
Any chance we can avoid the for-loop here? Something like `out += self.biases[type_vec]` should work as well.